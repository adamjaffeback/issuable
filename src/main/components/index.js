export { default as Layout } from './Layout';
export { default as Routes } from './Routes';
