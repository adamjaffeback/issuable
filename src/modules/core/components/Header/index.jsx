import React from 'react';

import { classie } from 'utils';
import Button from '../Button';

const Header = ({ children, className }) => (
  <header
    className={classie(['header', className])}>
    <h4 className="page-title">Issuable</h4>

    <Button>Logout</Button>
  </header>
);

export default Header;
