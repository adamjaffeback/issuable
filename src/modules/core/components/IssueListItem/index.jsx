import React from 'react';
import moment from 'moment';

import { classie } from 'utils';

import fallbackImg from 'assets/images/unassigned.png';

const DetailsLine = ({ title, created, updated }) => (
  <div className="details">
    {title}
    <br />
    <span className="times">
      created {created} â€¢ updated {updated}
    </span>
  </div>
);

class IssueListItem extends React.Component {
  handleKeyDown = (evt) => {
    const { onKeyDown, value } = this.props;
    const { key } = evt;
    evt.preventDefault();
    onKeyDown(key, value);
  };

  handleSelect = () => {
    const { onSelect, value } = this.props;
    onSelect(value);
  };

  handleBlur = () => {
    const { onSelect } = this.props;
    onSelect(null);
  };

  render() {
    const {
      option,
      value,
      selectedValue,
    } = this.props;

    const {
      assigneeAvatar,
      assignee,
      title,
      createdAt,
      updatedAt,
    } = option.data;

    const created = moment(createdAt).format('MM/DD/YYYY');
    const updated = moment(updatedAt).fromNow();

    return (
      <li
        tabIndex={1}
        onKeyDown={this.handleKeyDown}
        onClick={this.handleSelect}
        onBlur={this.handleBlur}
        className={classie(['selectListItem', 'issueListItem'], {
          selected: selectedValue === value,
        })}>
        <div className="imgWrap">
          <img src={assigneeAvatar || fallbackImg} alt={assignee} />
        </div>

        <DetailsLine
          title={title}
          created={created}
          updated={updated} />
      </li>
    );
  }
}

export default IssueListItem;
