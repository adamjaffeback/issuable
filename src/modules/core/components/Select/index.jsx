import React, { Component } from 'react';

import { classie } from 'utils';

import './styles.scss';

const Option = ({ value, label, className, hidden }) => (
  <option
    hidden={hidden}
    className={classie(['option', className])}
    value={value}>
    {label}
  </option>
);

class Select extends Component {
  static defaultProps = {
    onChange: () => null,
    value: {},
    options: [],
    showPlaceholder: true,
    placeholder: 'Select',
  };

  handleChange = (evt) => {
    const { onChange } = this.props;
    const { target: { value, name } } = evt;
    onChange(value, name);
  };

  render() {
    const {
      value,
      options,
      name,
      className,
      optionClassName,
      showPlaceholder,
      placeholder,
    } = this.props;

    return (
      <select
        value={value || ''}
        name={name}
        onChange={this.handleChange}
        className={classie(['select', className])}>
        {showPlaceholder &&
          <Option
            hidden={true}
            label={placeholder}
            className={optionClassName}
            value="" />}

        {options.map((opt, idx) => (
          <Option
            value={opt.value}
            key={`${name}-option-${idx}`}
            label={opt.label}
            className={optionClassName} />
        ))}
      </select>
    );
  }
}

export default Select;
