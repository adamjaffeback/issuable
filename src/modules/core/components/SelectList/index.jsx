import React, { Component, PureComponent } from 'react';

import { classie } from 'utils';

class SelectListItem extends PureComponent {
  handleClick = () => {
    const { value, onSelect } = this.props;
    onSelect(value);
  };

  render() {
    const {
      label,
      className,
    } = this.props;

    return (
      <li
        onClick={this.handleClick}
        className={classie(['selectListItem', className])}>
        {label}
      </li>
    );
  }
}

class SelectList extends Component {
  static defaultProps = {
    onSelection: () => null,
    value: {},
    options: [],
    onItemKeyDown: () => null,
  };

  handleSelection = (value) => {
    const { onSelection, name } = this.props;
    onSelection(value, name);
  };

  renderItem = (opt, idx) => {
    const {
      OptionComponent,
      optionClassName,
      value,
      onItemKeyDown,
    } = this.props;

    if (OptionComponent) {
      return (
        <OptionComponent
          onKeyDown={onItemKeyDown}
          key={opt.value}
          selectedValue={value}
          option={opt}
          index={idx}
          value={opt.value}
          onSelect={this.handleSelection} />
      );
    }

    return (
      <SelectListItem
        onSelect={this.handleSelection}
        value={opt.value}
        key={value}
        label={opt.label}
        className={classie([optionClassName], {
          selected: value === opt.value,
        })}
      />
    );
  };

  render() {
    const {
      options,
      className,
    } = this.props;

    return (
      <ul
        className={classie(['selectList', className])}>
        {options.map(this.renderItem)}
      </ul>
    );
  }
}

export default SelectList;
