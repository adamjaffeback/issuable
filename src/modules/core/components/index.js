export { default as Button } from './Button';
export { default as EmptyState } from './EmptyState';
export { default as CardPanel } from './CardPanel';
export { default as Header } from './Header';
export { default as IssueListItem } from './IssueListItem';
export { default as Select } from './Select';
export { default as SelectList } from './SelectList';
export { default as TextField } from './TextField';
