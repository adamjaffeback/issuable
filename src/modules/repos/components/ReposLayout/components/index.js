export { default as Issues } from './Issues';
export { default as RepoSelect } from './RepoSelect';
