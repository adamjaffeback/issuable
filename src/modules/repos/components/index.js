export { default as RepoSelect } from './RepoSelect';
