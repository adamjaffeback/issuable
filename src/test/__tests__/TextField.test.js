import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { TextField } from '../../modules/core/components';

describe('<TextField />', () => {
  it('Renders with empty string default value', () => {
    const renderedValue = renderer.create(
      <TextField />,
    );

    expect(renderedValue).toMatchSnapshot();
  });

  it('Applies its `name` prop', () => {
    const renderedValue = renderer.create(
      <TextField name="fullName" />
    );

    expect(renderedValue).toMatchSnapshot();
  });

  it('Applies its `value` prop', () => {
    const renderedValue = renderer.create(
      <TextField value="Teri Dactyl" />
    );

    expect(renderedValue).toMatchSnapshot();
  });

  it('Applies className prop to class attribute', () => {
    const item = shallow(
      <TextField className="added" />,
    );

    expect(item.find('.textField.added').exists()).toEqual(true);
  });

  it('simulates change events', () => {
    const onFieldChange = jest.fn();
    const event = { target: { value: 'Teri Dactyl', name: 'fullName' } };

    const item = shallow(
      <TextField
        name="fullName"
        value={''}
        onChange={onFieldChange} />,
    );

    item.find('input').simulate('change', event);

    expect(onFieldChange.mock.calls.length).toBe(1);
  });
});
