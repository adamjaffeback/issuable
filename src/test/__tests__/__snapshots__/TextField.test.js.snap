// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> Applies its \`name\` prop 1`] = `
<input
  className="textField"
  name="fullName"
  onKeyDown={[Function]}
  type="text"
  value=""
/>
`;

exports[`<TextField /> Applies its \`value\` prop 1`] = `
<input
  className="textField"
  onKeyDown={[Function]}
  type="text"
  value="Teri Dactyl"
/>
`;

exports[`<TextField /> Renders with empty string default value 1`] = `
<input
  className="textField"
  onKeyDown={[Function]}
  type="text"
  value=""
/>
`;
